src/Components/Endpoints/src/FormMapping/Converters/FileConverter.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components.Endpoints.FormMapping;
+
+internal sealed class FileConverter<T> : FormDataConverter<T>, ISingleValueConverter
+{
+    [RequiresDynamicCode(FormMappingHelpers.RequiresDynamicCodeMessage)]
+    [RequiresUnreferencedCode(FormMappingHelpers.RequiresUnreferencedCodeMessage)]
+    internal override bool TryRead(ref FormDataReader reader, Type type, FormDataMapperOptions options, out T? result, out bool found)
+    {
+        if (reader.FormFileCollection == null)
+        {
+            result = default;
+            found = false;
+            return true;
+        }
+
+        if (typeof(T) == typeof(IFormFileCollection))
+        {
+            result = (T)reader.FormFileCollection;
+            found = true;
+            return true;
+        }
+
+        var formFileCollection = reader.FormFileCollection;
+        if (formFileCollection.Count == 0)
+        {
+            result = default;
+            found = false;
+            return true;
+        }
+
+        var file = formFileCollection.GetFile(reader.CurrentPrefix.ToString());
```

Comment:
>     public IFormFileCollection Reports { get; set; }

This one is interesting. At the moment, minimal APIs assumes that a top-level `IFormFileCollection` type refers to `HttpContext.Request.Form.Files` in all occasions. We don't use the parameter name to extract a subset of files by their name. IMO, it probably makes more sense to support that by supporting an `IReadOnlyList<IFormFile>` type similar to what `IFormFIleCollection.GetFiles` returns. 

Binding for multiple `IFormFile`s with different names and erroring for multiple files with the same name makes sense though.

Reply:
@captainsafia I'm ok if we keep `IFormFileCollection` binding the entire set of files independent of the prefix (the type acts as the signal/discriminator) of what you want, and it's a well-defined behavior.

What I am interested in making sure that we do correctly is handling `input type=file` and `input type=file multiple` in a way that people can either use the inputs directly or rely on `InputFile` to author the UI.

We need to get this right because whatever we bake in now, will become a breaking change in the future otherwise. So I think your proposal of having IFormFileCollection bind to the entire thing and have `<<(Collection|Dictionary)Type>><IFormFile>` bind to the set that match the proposed subset makes sense.

Reply:
OK, I've updated the PR to accommodate some of the functionality here. InputFile works with:

- IBrowserFile
- IFormFile
- IReadOnlyList<IFormFile>
- IFormFileCollection

as types backing the property.

> Binding for multiple IFormFiles with different names and erroring for multiple files with the same name makes sense though.

I take this back. It appears the current behavior for `IFormFileCollection` is to return the _first_ file with a given name if they are multiple so I kept this pattern in the implementation.